src/acl/ACL.ts

```diff
@@ -130,8 +130,16 @@ class ACL {
     nodeId2: NodeId,
   ): Promise<boolean> {
     return await this._transaction(async () => {
-      const permId1 = await this.db.get(this.aclNodesDbDomain, nodeId1, true);
-      const permId2 = await this.db.get(this.aclNodesDbDomain, nodeId2, true);
+      const permId1 = await this.db.get(
+        this.aclNodesDbDomain,
+        nodeId1.toBuffer(),
```

Comment:
The `aclNodesDb` is `NodeId -> PermissionId`. You should ensure that both the key and value are raw buffers, and the `raw` option is set to true here so that the DB doesn't do any conversion/transcoding of the buffers. The `raw` parameter is the third parameter. It ensures that you return a `Buffer` directly.

Also this assumes that `PermId` is also transitioned to js-id as well, so it is also a buffer, however if that isn't the case, disregard this comment. But this is relevant to #299.

Reply:
`PermId` is an js-id and in that example is already being used as you described. same for all other times we get the PermId from the db.

Reply:
Is `PermId` being encoded as JSON or as raw buffer when stored as a value in the DB?

Reply:
Looks like it's getting stored as a buffer.

Reply:
Ok great. We bear in the mind that if is ever wrapped in JSON we have to use 3.3.0's ability of `fromJSON` and the `reviver` parameter.